tests/test_nominatim_gazetteer.py

```
@@ -0,0 +1,215 @@
+        "osm_id": osm_id,
+        "type": type_,
+    }
+    return loc
+
+
+@pytest.fixture
+def fake_client(monkeypatch):
+    """Replace NominatimGazetteer's geopy client with a controllable mock."""
+    client = MagicMock()
+    monkeypatch.setattr(ng.NominatimGazetteer, "_build_client",
+                        lambda self: client)
```

Ruff (0.15.12)

[warning] Unused lambda argument: `self`

(ARG005)